gateway/host_checker_manager_test.go

```diff
@@ -0,0 +1,183 @@
+package gateway
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/TykTechnologies/tyk/apidef"
+	"github.com/TykTechnologies/tyk/config"
+	"github.com/TykTechnologies/tyk/storage"
+	uuid "github.com/satori/go.uuid"
+)
+
+func TestHostCheckerManagerInit(t *testing.T) {
+
+	hc := HostCheckerManager{}
+	redisStorage := &storage.RedisCluster{KeyPrefix: "host-checker-test:"}
+	hc.Init(redisStorage)
+
+	if hc.Id == "" {
+		t.Error("HostCheckerManager should create an Id on Init")
+	}
+	if hc.unhealthyHostList == nil {
+		t.Error("HostCheckerManager should initialize unhealthyHostList on Init")
+	}
+	if hc.resetsInitiated == nil {
+		t.Error("HostCheckerManager should initialize resetsInitiated on Init")
+	}
+}
+
+func TestAmIPolling(t *testing.T) {
+	hc := HostCheckerManager{}
+
+	polling := hc.AmIPolling()
+	if polling {
+		t.Error("HostCheckerManager storage not configured, it should have failed.")
+	}
+
+	//Testing if we had 2 active host checker managers, only 1 takes control of the uptimechecks
+	globalConf := config.Global()
+	globalConf.UptimeTests.PollerGroup = "TEST"
+	config.SetGlobal(globalConf)
+	defer ResetTestConfig()
+
+	redisStorage := &storage.RedisCluster{KeyPrefix: "host-checker-test:"}
+	hc.Init(redisStorage)
+	hc2 := HostCheckerManager{}
+	hc2.Init(redisStorage)
+
+	polling = hc.AmIPolling()
+	defer hc.StopPoller()
+
+	pollingHc2 := hc2.AmIPolling()
+	defer hc2.StopPoller()
+
+	if !polling && pollingHc2 {
+		t.Error("HostCheckerManager storage configured, it shouldn't have failed.")
+	}
+
+	//Testing if the PollerCacheKey contains the poller_group
+	activeInstance, err := hc.store.GetKey("PollerActiveInstanceID.TEST")
+	if err != nil {
+		t.Error("PollerActiveInstanceID.TEST  should exist in redis.", activeInstance)
+	}
+	if activeInstance != hc.Id {
+		t.Error("PollerActiveInstanceID.TEST value should be hc.Id")
+	}
+
+	ResetTestConfig()
+	//Testing if the PollerCacheKey doesn't contains the poller_group by default
+	hc = HostCheckerManager{}
+	redisStorage = &storage.RedisCluster{KeyPrefix: "host-checker-test:"}
+	hc.Init(redisStorage)
+	hc.AmIPolling()
+	defer hc.StopPoller()
+
+	activeInstance, err = hc.store.GetKey("PollerActiveInstanceID")
+	if err != nil {
+		t.Error("PollerActiveInstanceID should exist in redis.", activeInstance)
+	}
+	if activeInstance != hc.Id {
+		t.Error("PollerActiveInstanceID value should be hc.Id")
+	}
+
+}
```

Comment:
We are hardcoding `PollerActiveInstanceID` and also the errors  lack  more context.

We should have something like 

```diff
diff --git a/gateway/host_checker_manager_test.go b/gateway/host_checker_manager_test.go
index c320147e..cb1dca59 100644
--- a/gateway/host_checker_manager_test.go
+++ b/gateway/host_checker_manager_test.go
@@ -38,7 +38,8 @@ func TestAmIPolling(t *testing.T) {
 
 	//Testing if we had 2 active host checker managers, only 1 takes control of the uptimechecks
 	globalConf := config.Global()
-	globalConf.UptimeTests.PollerGroup = "TEST"
+	groupID := "TEST"
+	globalConf.UptimeTests.PollerGroup = groupID
 	config.SetGlobal(globalConf)
 	defer ResetTestConfig()
 
@@ -58,12 +59,13 @@ func TestAmIPolling(t *testing.T) {
 	}
 
 	//Testing if the PollerCacheKey contains the poller_group
-	activeInstance, err := hc.store.GetKey("PollerActiveInstanceID.TEST")
+	testKey := PollerCacheKey + "." + groupID
+	activeInstance, err := hc.store.GetKey(testKey)
 	if err != nil {
-		t.Error("PollerActiveInstanceID.TEST  should exist in redis.", activeInstance)
+		t.Errorf("%q  should exist in redis.%v", testKey, activeInstance)
 	}
 	if activeInstance != hc.Id {
-		t.Error("PollerActiveInstanceID.TEST value should be hc.Id")
+		t.Errorf("%q value : expected %v got %v", testKey, hc.Id, activeInstance)
 	}
 
 	ResetTestConfig()
@@ -74,12 +76,12 @@ func TestAmIPolling(t *testing.T) {
 	hc.AmIPolling()
 	defer hc.StopPoller()
 
-	activeInstance, err = hc.store.GetKey("PollerActiveInstanceID")
+	activeInstance, err = hc.store.GetKey(PollerCacheKey)
 	if err != nil {
-		t.Error("PollerActiveInstanceID should exist in redis.", activeInstance)
+		t.Errorf("%q should exist in redis.%v", PollerCacheKey, activeInstance)
 	}
 	if activeInstance != hc.Id {
-		t.Error("PollerActiveInstanceID value should be hc.Id")
+		t.Errorf("%q : value expected %v got %v", PollerCacheKey, hc.Id, activeInstance)
 	}
 
 }
```